ui/components/multichain/dropdown-editor/dropdown-editor.tsx

```diff
@@ -0,0 +1,260 @@
+  onItemAdd,
+  onDropdownOpened,
+  itemKey,
+  itemIsDeletable = () => true,
+  renderItem,
+  renderTooltip,
+  buttonDataTestId,
+}: {
+  title: string;
+  placeholder: string;
+  items?: Item[];
+  selectedItemIndex?: number;
+  addButtonText: string;
+  error?: boolean;
+  style: DropdownEditorStyle;
+  onItemSelected: (index: number) => void;
+  onItemDeleted: (deletedIndex: number, newSelectedIndex?: number) => void;
+  onItemAdd: () => void;
+  onDropdownOpened?: () => void;
+  itemKey: (item: Item) => string;
+  itemIsDeletable?: (item: Item, items: Item[]) => boolean;
+  renderItem: (item: Item, isList: boolean) => string | ReactNode;
+  renderTooltip: (item: Item, isList: boolean) => string | undefined;
+  buttonDataTestId: string;
+}) => {
+  const t = useI18nContext();
+  const dropdown = useRef(null);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const renderDropdownList = () => (
+    <Box>
+      {items?.map((item, index) => {
+        const row = (
+          <Box
+            alignItems={AlignItems.center}
+            paddingLeft={4}
+            paddingRight={4}
+            display={Display.Flex}
+            justifyContent={JustifyContent.spaceBetween}
+            key={itemKey(item)}
+            onClick={() => {
+              onItemSelected(index);
+              setIsDropdownOpen(false);
+            }}
+            className={classnames('dropdown-editor__item', {
+              'dropdown-editor__item--selected': index === selectedItemIndex,
+            })}
+          >
+            {index === selectedItemIndex && (
+              <Box
+                className="dropdown-editor__item-selected-pill"
+                borderRadius={BorderRadius.pill}
+                backgroundColor={BackgroundColor.primaryDefault}
+              />
+            )}
+            {renderItem(item, true)}
+            {itemIsDeletable(item, items) && (
+              <ButtonIcon
+                marginLeft={1}
+                ariaLabel={t('delete')}
+                size={ButtonIconSize.Sm}
+                iconName={IconName.Trash}
+                color={IconColor.errorDefault}
+                onClick={(e: React.MouseEvent) => {
+                  e.stopPropagation();
+
+                  // Determine which item should be selected after deletion
+                  let newSelectedIndex;
+                  if (selectedItemIndex === undefined || items.length <= 1) {
+                    newSelectedIndex = undefined;
+                  } else if (index === selectedItemIndex) {
+                    newSelectedIndex = 0;
+                  } else if (index > selectedItemIndex) {
+                    newSelectedIndex = selectedItemIndex;
+                  } else if (index < selectedItemIndex) {
+                    newSelectedIndex = selectedItemIndex - 1;
+                  }
+
+                  onItemDeleted(index, newSelectedIndex);
+                }}
+              />
+            )}
+          </Box>
+        );
```

Comment:
Not a dealbreaker, but for testing/debugging purposes I would typically prefer to have this live as it's own component, rather than in a callback within `dropdown-editor.tsx`. Maybe something like `DropdownList.tsx`, and then pass it whatever props it needs.

Doesn't need to block though.

Reply:
I agree!  Let's consider this as a follow up!